endoreg_db/serializers/video_segmentation.py

```
@@ -332,20 +369,48 @@ def save(self):
+        )  # Existing database segment keys
+        new_keys = set(
+            (float(seg["start_frame_number"]), float(seg["end_frame_number"]))
+            for seg in new_segments
+        )  # New frontend segment keys
+
+        # Start a transaction to ensure database consistency
         updated_segments = []
         new_entries = []
         existing_keys = set(existing_segments_dict.keys())
-        new_keys = set((float(seg["start_frame_number"]), float(seg["end_frame_number"])) for seg in new_segments)
+        new_keys = set(
```

Ruff (0.8.2)

Local variable `new_keys` is assigned to but never used

Remove assignment to unused variable `new_keys`

(F841)